pkg/pusher/pusher.go

```diff
@@ -311,10 +307,10 @@ func (s *Service) pushDirect(ctx context.Context, logger log.Logger, op *Op) err
 		}
 	}()
 
-	_, err = s.validStamp(op.Chunk)
-	if err != nil {
-		logger.Warning(
-			"stamp with is no longer valid, skipping direct upload for chunk",
+	ok, err := s.batchExist.Exists(op.Chunk.Stamp().BatchID())
```

Comment:
If batch not found it will return nil? Is this also ok?

Reply:
yes, for direct uploads, the chunk resync is retried if this error is not nil. because batch is no longer valid at this point, we should stop syncing, hence the nil return